src/main/java/gift/domain/embed/Wishlist.java

```diff
@@ -13,13 +14,20 @@
 public class Wishlist {
 
     @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, orphanRemoval = true)
-    private List<Wish> wishes;
+    private List<Wish> wishes = new ArrayList<>();
 
-    public Wishlist() {
-        this.wishes = new ArrayList<>();
+    public Wishlist() {}
+
+    public List<Wish> getWishlist() {
+        return Collections.unmodifiableList(new ArrayList<>(wishes));
     }
 
     public void add(Wish wish) {
+        if (wishes.stream()
+            .anyMatch(elem -> elem.getProduct().getId().equals(wish.getProduct().getId()))) {
```

Comment:
```suggestion
            .anyMatch(elem -> elem.equals(wish))) {
```
- 디미터 원칙 위반이네요. 객체에 메시지를 보내보는건 어떨까요? 위와 같이 구현하려면, 각각 Wish(동등성), Product(동일성) 규칙에 따라 Eqauls를 구현하면 되겠죠? 다른 부분들도 한번 리팩토링해보세요

Reply:
- 반환값의 메서드를 바로 호출하는 것은 좋지 않다
- 객체의 필드에 직접 접근하는 것도 좋지 않다
- 객체는 자신의 상태를 외부에서 알 수 없도록 감춰야 한다
- `객체지향 생활체조`에서는 `한 줄에 점을 하나만 찍는다`로 요약된다
> 결합도를 낮춰, 유연하고 유지보수성 높은 코드를 작성하기 위한 규칙!

덕분에 `디미터 원칙`에 대해 자세히 정리해볼 수 있었습니다.

해당 코드 외에도 필드의 메서드가 호출되던 부분들을 찾아 수정해줬습니다!